subsys/pm/device_runtime.c

```diff
@@ -127,71 +128,93 @@ static int pm_device_request(const struct device *dev,
 						  PM_DEVICE_STATE_SUSPEND,
 						  NULL, NULL);
 		}
-		ret = 0;
 		goto out;
 	}
 
-	(void)k_work_schedule(&dev->pm->work, K_NO_WAIT);
+	(void)k_mutex_lock(&dev->pm->lock, K_FOREVER);
+
+	if (!dev->pm->enable) {
+		ret = -ENOTSUP;
+		goto out_unlock;
+	}
+
+	if (target_state == PM_DEVICE_STATE_ACTIVE) {
+		dev->pm->usage++;
+		if (dev->pm->usage < 0) {
```

Comment:
can you add some comments on this condition? When would the usage be negative after incrementing it?

Reply:
That is just because the old implementation was allowing the reference count gets negative due to unbalanced pm_device_put() calls. I just kept this behavior for now.

Reply:
I think it is fine removing it here though. Just did it.

Reply:
I'm not clear on this part neither. Do you plan an update ?